src/network/mod.rs

```diff
@@ -224,25 +231,24 @@ impl GameSniffer {
         }
 
         if let Some(kcp) = kcp {
-            let commands = kcp
-                .receive_segments(kcp_seg)
-                .into_iter()
-                .filter_map(|data| self.receive_command(data))
-                .collect();
+            let mut commands = vec![];
+            for command in kcp.receive_segments(kcp_seg)? {
+                commands.push(self.receive_command(command)?);
+            }
 
-            return Some(commands);
+            return Ok(commands);
         }
 
-        None
+        anyhow::bail!("fail ti read kcp segment")
     }
 
     #[instrument(skip_all, fields(len = data.len()))]
-    fn receive_command(&mut self, mut data: Vec<u8>) -> Option<GameCommand> {
+    fn receive_command(&mut self, mut data: Vec<u8>) -> Result<GameCommand> {
         let key = match &self.key {
             Some(k) => k,
             None => {
-                self.key = Some(lookup_initial_key(&self.initial_keys, &data));
-                self.key.as_ref().unwrap()
+                self.key = Some(lookup_initial_key(&self.initial_keys, &data)?);
+                self.key.as_ref().context("this should not happens")?
```

Comment:
```suggestion
                self.key.as_ref().context("this should not happen")?
```

`expect` is fine if it really can't happen, though.